### PR TITLE
Netfile proof of concept

### PR DESCRIPTION
**Pull request by adborden**
Fetches transactions for 2014 via [`/public/campaign/export/cal201/transaction/year`](https://netfile.com/Connect2/api/swagger-ui/#!/public/postpubliccampaignexportcal201transactionyear) for all the agencies on Netfile and dumps them as csvs in `./netfile_raw_data`. The idea is to then map the different transaction types to what `calaccess_raw` already uses, then we just use the same models/code to load the data. I _think_ this means mapping the [transaction types](https://netfile.com/Connect2/api/swagger-ui/#!/public/getpubliccampaignlisttransactiontypes) to Django models, but I'm not sure. I could use some help with this.

Notes:
- Not all agencies have data for this endpoint.
- I'm limiting the number of fetches to Netfile (max 2 pages ~2000 transactions).
- Data downloaded is about 21M right now (again, limited).
- Also, I'm trying to level up my Django-fu, so let me know if you have any tips :)
- You'll want to `pip install -r requirements_dev.txt` to pull in the swagger-py module

Here's a list of the fields we get from this endpoint:
- filerid
- dist_no
- end_bal
- amounttype
- tran_zip4
- tran_id
- loan_date1
- loan_date2
- tran_naml
- tran_namf
- tran_nams
- tran_namt
- tran_dscr
- intr_emp
- tran_city
- tran_amt1
- g_from_e_f
- cmte_id
- loan_amt8
- amt_incur
- filingid
- bakref_tid
- loan_amt2
- loan_amt3
- tres_zip4
- loan_amt6
- loan_amt7
- loan_amt4
- loan_amt5up_opp_cd
- memo_code
- intr_adr2
- intr_namt
- intr_nams
- intr_naml
- externalid
- intr_namf
- transactiontype
- tres_naml
- calculated_amount
- tres_namf
- intr_self
- tres_nams
- tres_namt
- tran_emp
- tran_adr2
- tres_st
- intr_st
- intr_zip4
- intr_adr1
- elec_date
- tran_amt2
- xref_schnum
- xref_match
- int_rate
- juris_cd
- tran_code
- off_s_h_cd
- office_cd
- bal_juris
- loan_amt1
- tran_chkno
- lender_name
- form_type
- tran_date
- latitude
- tran_type
- longitude
- rec_type
- intr_city
- tran_adr1
- int_cmteid
- calculated_date
- bal_name
- office_dscr
- loan_rate
- filername
- beg_bal
- memo_refno
- tran_occ
- tran_date1
- amt_paid
- entity_cd
- tres_adr1
- tres_adr2
- intr_occ
- netfilekey
- bal_num
- cand_namf
- cand_naml
- cand_nams
- cand_namt
- tres_city
- juris_dscr
- tran_self
- tran_st

### CODE REVIEW

File: netfile/connect2_api.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+        })
+
+    # Transactions in SF
+    transaction = transactions.next() # Work-around generator magic
+    headers = transaction.keys()
+    transaction_csv = csv.DictWriter(csvfile, headers)
+    transaction_csv.writeheader()
+    for transaction in transactions:
+        transaction_csv.writerow(transaction)
+    return
```
Comment (tdooner) [inline netfile/connect2_api.py]: Did you mean to leave this there?

**Comment by tdooner** [comment]
Sweet. :+1: Unfortunately my django fu is pretty limited so I don't have any advice. Perhaps someone else does. Anyway, I think you should be the one to hit the merge button.

**Comment by rkiddy** [comment]
Looks to be a great start. It does not work for me at the moment, but I assume that is mostly me. A few things, though.

1) grep -r pubic *

These are just in doc lines, but there they are.

2) I am not seeing swaggerpy in the requirements.txt. Does that need to be updated? I installed it myself to get past the first error.

3) Any sugestions on this would be appreciated. I am running on Ubuntu 14.04.

```
 $ python example/manage.py downloadnetfilerawdata 
 Traceback (most recent call last):
   File "example/manage.py", line 9, in <module>
     execute_from_command_line(sys.argv)
   File "/usr/local/lib/python2.7/dist-packages/django/core/management/__init__.py", line 385, in execute_from_command_line
     utility.execute()
   File "/usr/local/lib/python2.7/dist-packages/django/core/management/__init__.py", line 377, in execute
     self.fetch_command(subcommand).run_from_argv(self.argv)
   File "/usr/local/lib/python2.7/dist-packages/django/core/management/base.py", line 288, in run_from_argv
     self.execute(*args, **options.__dict__)
   File "/usr/local/lib/python2.7/dist-packages/django/core/management/base.py", line 338, in execute
     output = self.handle(*args, **options)
   File "netfile/management/commands/downloadnetfilerawdata.py", line 38, in handle
     self.connect2 = Connect2API()
   File "netfile/connect2_api.py", line 57, in __init__
     self.api = SwaggerClient(CONNECT2_SPEC)
   File "/usr/local/lib/python2.7/dist-packages/swaggerpy/client.py", line 178, in __init__
     self.api_docs = loader.load_resource_listing(url_or_resource)
   File "/usr/local/lib/python2.7/dist-packages/swaggerpy/swagger_model.py", line 173, in load_resource_listing
     self.process_resource_listing(resource_listing)
   File "/usr/local/lib/python2.7/dist-packages/swaggerpy/swagger_model.py", line 197, in process_resource_listing
     processor.apply(resources)
   File "/usr/local/lib/python2.7/dist-packages/swaggerpy/processors.py", line 112, in apply
     self.process_api_declaration(**context.args)
   File "/usr/local/lib/python2.7/dist-packages/swaggerpy/swagger_model.py", line 56, in process_api_declaration
     validate_required_fields(resource, required_fields, context)
   File "/usr/local/lib/python2.7/dist-packages/swaggerpy/swagger_model.py", line 213, in validate_required_fields
     "Missing fields: %s" % ', '.join(missing_fields), context)
 swaggerpy.processors.SwaggerError: ('Missing fields: swaggerVersion', ParsingContext(stack=['resources=https://netfile.com/Connect2/api/resources', u'resource=https://netfile.com/Connect2/api/resource/auth']), None)
```

**Comment by adborden** [comment]
@rkiddy you'll need a special fork of swagger-py, but it should be in the [requirements_dev.txt](https://github.com/opencalifornia/disclosure-backend/pull/4/files#diff-9746485032a333bedc47e790ce5228d8R13). Try:

```
$ pip uninstall swaggerpy; pip install -r requirements_dev.txt
```

**Comment by adborden** [comment]
Hmm, hang tight, Travis is choking on the requirements too, I think I need to switch it to an https url.

**Comment by adborden** [comment]
@rkiddy okay, try now.

```
$ pip uninstall swaggerpy; pip install -r requirements_dev.txt
```

**Pull request merged**